python_modules/dagster/dagster/_core/host_representation/external_data.py

```diff
@@ -1659,6 +1662,50 @@ def external_asset_graph_from_defs(
             )
         )
 
+    # resolve deps unexec assets since they are not encoded in jobs
+    unexecutable_keys = set()
+    for unexec_def in unexecutable_assets_defs or []:
+        for key in unexec_def.keys:
+            unexecutable_keys.add(key)
+            for depends_on in unexec_def.asset_deps[key]:
+                deps[key][depends_on] = ExternalAssetDependency(depends_on)
+                dep_by[depends_on][key] = ExternalAssetDependedBy(key)
+
+    # build nodes for unexecutable assets defs
+    for unexec_def in unexecutable_assets_defs or []:
+        for key in unexec_def.keys:
+            asset_nodes.append(
+                ExternalAssetNode(
+                    asset_key=key,
+                    dependencies=list(deps[key].values()),
+                    depended_by=list(dep_by[key].values()),
+                    metadata=unexec_def.metadata_by_key[key],
+                    group_name=unexec_def.group_names_by_key[key],
+                    is_source=False,
+                    is_observable=False,
+                    # could use a special kind tag for more UI punch
+                    # compute_kind=...
+                )
+            )
+
+    # build nodes for remaining asset keys that were referred in other assets but not defined
+    asset_keys_without_definitions = (
+        all_upstream_asset_keys.difference(node_defs_by_asset_key.keys())
+        .difference(source_assets_by_key.keys())
+        .difference(unexecutable_keys)
+    )
+    for asset_key in asset_keys_without_definitions:
+        asset_nodes.append(
+            ExternalAssetNode(
+                asset_key=asset_key,
+                dependencies=list(deps[asset_key].values()),
+                depended_by=list(dep_by[asset_key].values()),
+                job_names=[],
+                group_name=group_name_by_asset_key.get(asset_key),
+                code_version=code_version_by_asset_key.get(asset_key),
+            )
+        )
```

Comment:
this part of the building (moved from above) dates back to the addition of `ExternalAssetNode` https://github.com/dagster-io/dagster/pull/4742/files

not sure what conditions it fires with the source asset resolution happening above